spatialos-sdk/src/worker/schema.rs

```diff
@@ -0,0 +1,606 @@
+use crate::worker::{internal::SchemaFieldSealed, EntityId};
+use spatialos_sdk_sys::worker::*;
+use std::{collections::BTreeMap, mem, slice};
+
+mod component_data;
+mod component_update;
+mod object;
+pub mod owned;
+
+pub use self::{component_data::*, component_update::*, object::*};
+
+pub type FieldId = u32;
+
+// =================================================================================================
+// Schema Conversion Traits
+// =================================================================================================
+
+pub trait SchemaField: Sized + SchemaFieldSealed {
+    type RustType: Sized;
+
+    fn get_field(object: &Object, field: FieldId) -> Self::RustType;
+
+    fn add_field(object: &mut Object, field: FieldId, value: &Self::RustType);
+
+    fn has_update(update: &ComponentUpdate, field: FieldId) -> bool;
+
+    fn get_update(update: &ComponentUpdate, field: FieldId) -> Option<Self::RustType> {
```

Comment:
I find `has_update`, `get_update` and `add_update` to be quite confusing at first. Perhaps they should be called `update_has_field`, `update_get_field` and `update_add_field`?